### DISCUSSION TITLE
protect from killactive?

### DISCUSSION THREAD

**Discussion by bobblkabb** [hyprwm/Hyprland#2317] [Q&A]
Got a terminal in a special workspace - like a yakuake/guake terminal and sometimes I accidentally close it (killactive).
Is there a way to protect from killing it?

**Comment by NotAShelf** [comment]
I don't think there is a windowrole that protects a window from being killed, but it's probably scriptable.
You can alias `hyprctl kill` (or make a plugin) and check a window's class/title before attempting to kill it.

**Reply by vaxerski** [reply]
he's talking about the dispatcher not the hyprctl command. In that case you can just parse `hyprctl activewindow` and do your logic

**Comment by Frestein** [comment]
```sh
#!/bin/sh

LOCK_FILE="/tmp/killactive.lock"
TIMEOUT=300

if [ -f "$LOCK_FILE" ]; then
	rm -f "$LOCK_FILE"
	hyprctl dispatch killactive
	exit 0
fi

touch "$LOCK_FILE"
(
	sleep "$(echo "$TIMEOUT / 1000" | bc -l)"
	rm -f "$LOCK_FILE"
) &
```
